extractor/src/extract/query_builder.py

```diff
@@ -131,9 +144,14 @@ def add_column_to_query(self, sql_column: SqlColumn, sqlalchemy_table: Table, qu
         sqlalchemy_col = self.get_column(sql_column, sqlalchemy_table)
 
         # Add the column to the _cur_query_columns attribute
-        self._cur_query_columns.add(sql_column)
+        self._cur_query_columns[sql_column] = sqlalchemy_col
 
-        return query.add_columns(sqlalchemy_col)
+        # We aggregate the rows from foreign tables in arrays
+        return query.add_columns(
+            sqlalchemy_col
+            if sql_column.table_name() == self.analysis.primary_key_column.table_name()
```

Comment:
is this the same condition that the one on line 94 ? if yes, maybe we can add a method on the sqlcolumn class

Reply:
That's right, it's the same! Do you think it's really worth adding a method?

Reply:
no, you're right